packages/fiori/src/TimelineGroupItem.ts

```diff
@@ -0,0 +1,191 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import ToggleButton from "@ui5/webcomponents/dist/ToggleButton.js";
+import TimelineLayout from "./types/TimelineLayout.js";
+import TimelineItem from "./TimelineItem.js";
+import type { ITimelineItem } from "./Timeline.js";
+
+import TimelineGroupItemTemplate from "./generated/templates/TimelineGroupItemTemplate.lit.js";
+
+// Styles
+import TimelineGroupItemCss from "./generated/themes/TimelineGroupItem.css.js";
+
+const SHORT_LINE_WIDTH = "ShortLineWidth";
+const LARGE_LINE_WIDTH = "LargeLineWidth";
+
+type TimelineGroupItemToggleCollapseEventDetail = {
+	collapsed: boolean;
+};
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * An entry posted on the timeline.
+ * It is intented to represent a group of `<ui5-timeline-item>`s.
+ *
+ * **Note**: Please do not use empty groups in order to preserve the intended design.
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 2.1.0
+ */
+@customElement({
+	tag: "ui5-timeline-group-item",
+	renderer: litRender,
+	styles: TimelineGroupItemCss,
+	template: TimelineGroupItemTemplate,
+	dependencies: [TimelineItem, ToggleButton],
+})
+/**
+ * Fired when the group item is expanded or collapsed.
+ * @public
+ * @param {boolean} collapsed Indicator whether the group item is collapsed or expanded.
+ */
+@event<TimelineGroupItemToggleCollapseEventDetail>("toggle-collapse", {
+	detail: {
+		collapsed: { type: Boolean },
+	},
+})
+class TimelineGroupItem extends UI5Element implements ITimelineItem {
+	/**
+	 * Defines the text of the button that expands and collapses the group.
+	 * @public
+	 */
+	@property()
+	groupName = "";
+
+	/**
+	 * Determines if the group is collapsed or expanded.
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	set collapsed(value: boolean) {
+		const oldValue = this._collapsed;
```

Comment:
why do you need this setter? is it only to fire event? We fire event when user change state of the component using interaction. Probably we should fire this event somewhere else. If the event is fired somewhere else the parameter is redundant.